src/box/memtx_engine.cc

```diff
@@ -2032,18 +2033,20 @@ memtx_tuple_new_raw_impl(struct tuple_format *format, const char *data,
 		goto end;
 	}
 
-	while ((tuple = MemtxAllocator<ALLOC>::alloc_tuple(total)) == NULL) {
+	while ((block = MemtxAllocator<ALLOC>::alloc(tuple_len, data_offset,
+						     make_compact)) == NULL) {
 		bool stop;
 		memtx_engine_run_gc(memtx, &stop);
 		if (stop)
 			break;
 	}
-	if (tuple == NULL) {
+	if (block == NULL) {
 		diag_set(OutOfMemory, total, "slab allocator", "memtx_tuple");
 		goto end;
 	}
-	tuple_create(tuple, 0, tuple_format_id(format),
-		     data_offset, tuple_len, make_compact);
+
+	tuple = memtx_block_get_tuple(block);
```

Comment:
I'd name this method `memtx_block_to_tuple` for consistency with `memtx_block_from_tuple`. And I think we could rename `memtx_block_get_data` to `memtx_block_data`.